frontend/packages/dev-console/src/components/topology/components/componentUtils.ts

```diff
@@ -165,44 +171,45 @@ const nodeDropTargetSpec: DropTargetSpec<
   { canDrop: boolean; dropTarget: boolean; edgeDragging: boolean },
   NodeComponentProps
 > = {
-  accept: [MOVE_CONNECTOR_DROP_TYPE, CREATE_CONNECTOR_DROP_TYPE],
+  accept: [EDGE_DRAG_TYPE, CREATE_CONNECTOR_DROP_TYPE],
   canDrop: (item, monitor, props) => {
     if (isEdge(item)) {
-      return canDropEdgeOnNode(monitor.getOperation(), item, props.element);
+      return canDropEdgeOnNode(monitor.getOperation()?.type, item, props.element);
     }
     if (item === props.element) {
       return false;
     }
     return !props.element.getTargetEdges().find((e) => e.getSource() === item);
   },
   collect: (monitor, props) => ({
-    canDrop: highlightNode(monitor, props),
+    canDrop: highlightNode(monitor, props.element),
     dropTarget: monitor.isOver({ shallow: true }),
     edgeDragging: nodesEdgeIsDragging(monitor, props),
   }),
 };
 
-const graphWorkloadDropTargetSpec: DropTargetSpec<
-  GraphElement,
+const graphDropTargetSpec: DropTargetSpec<
+  GraphElement | DragNodeObject,
```

Comment:
I wonder if this was incorrectly typed before. Can we not make it:
```suggestion
  DragNodeObject,
```